libkbfs/folder_update_prepper.go

```diff
@@ -0,0 +1,1223 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libkbfs
+
+import (
+	"fmt"
+
+	"github.com/keybase/client/go/logger"
+	"github.com/keybase/client/go/protocol/keybase1"
+	"github.com/keybase/kbfs/kbfsblock"
+	"github.com/keybase/kbfs/tlf"
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+)
+
+// folderUpdatePrepper is a helper struct for preparing blocks and MD
+// updates before they get synced to the backend servers.  It can be
+// used for a single update or for a batch of updates (e.g. conflict
+// resolution).
+type folderUpdatePrepper struct {
+	config       Config
```

Comment:
Yeah I suppose instead of `testFooConfig` we could have
`newTestConfigLocalForFoo`, which populates only the fields that are needed.

I do like the explicit dependency specification, but I understand if you don't like the slight code overhead it adds per component.